From 28773ac06d0e0f00d6a56973eacae4f5df9e8cc2 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Sun, 16 Apr 2017 08:11:20 +0300
Subject: [PATCH] proc.h: add missing headers

ino_t and pid_t require stat.h and types.h, respectively. Fixes the following
build failure with musl libc:

In file included from cpp.c:7:0:
proc.h:11:31: error: unknown type name ‘ino_t’
 extern int proc_find_by_inode(ino_t ino, char *cmdline, size_t len, pid_t *pid);
                               ^
proc.h:11:69: error: unknown type name ‘pid_t’
 extern int proc_find_by_inode(ino_t ino, char *cmdline, size_t len, pid_t *pid);
                                                                     ^
proc.h:12:25: error: unknown type name ‘pid_t’
 extern bool proc_exists(pid_t pid);
                         ^

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---

Patch status: sent upstream
http://www.mail-archive.com/netsniff-ng%40googlegroups.com/msg01972.html

 proc.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/proc.h b/proc.h
index 4c34a6412a81..2ce0c6e7a932 100644
--- a/proc.h
+++ b/proc.h
@@ -2,6 +2,8 @@
 #define PROC_H
 
 #include <stdbool.h>
+#include <sys/stat.h>
+#include <sys/types.h>
 
 extern void cpu_affinity(int cpu);
 extern int set_proc_prio(int prio);
-- 
2.11.0

